lib/portlayer/storage/vsphere/volume.go

```diff
@@ -191,3 +193,57 @@ func (v *VolumeStore) VolumesList(op trace.Operation) ([]*storage.Volume, error)
 
 	return volumes, nil
 }
+
+// ReadArchive reads the delta between child and parent volume layers, returning
+// the difference as a tar archive.
+//
+// store - the volume store containing the two layers
+// child - the leaf layer of the volume chain
+// ancestor - the volume layer up the chain against which to diff
+// pathspec - describes filters on paths found in the data (include, exclude, strip)
+// data - set to true to include file data in the tar archive, false to include headers only
+func (v *VolumeStore) ReadArchive(op trace.Operation, store *url.URL, child, ancestor string, pathspec map[string]string, data bool) (io.ReadCloser, error) {
```

Comment:
I was thinking about this and I was wondering if you thought it might be better for us to pass in the `PathSpec`struct instead of the map for this function. Then we can export the `processPathSpec` This way we could `validate` a path spec so to speak before attempting a archive operation.